pepr/robustness/art_wrapper.py

```diff
@@ -0,0 +1,3106 @@
+"""PePR wrapper classes for ART attack classes."""
+
+import logging
+import numpy as np
+import os
+
+from pepr.attack import Attack
+from pepr import report
+import matplotlib.pyplot as plt
+from pylatex import Command, Tabular, MiniPage, NoEscape
+from pylatex.section import Subsubsection
+
+import art
+from art.estimators.classification import KerasClassifier
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+plt.style.use("default")
+# force line grid to be behind bar plots
+plt.rcParams["axes.axisbelow"] = True
+plt.rcParams["axes.grid"] = True
+plt.rcParams["grid.linestyle"] = ":"
+
+
+class BaseEvasionAttack(Attack):
+    """
+    Base ART attack class implementing the logic for running an evasion attack and
+    generating a report.
+
+    Parameters
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the class.
+    use_labels : bool
+        If true, the true labels are used as targets.
```

Comment:
Can we make this description a little more precise? Do you mean that the true labels are exploited for the attack?

Reply:
The ART documentation is not clearer. For some attacks you can pass the labels as target labels optionally. In my understanding, these target labels are the true labels. If they are not passed, then they are replaced by the target model's predictions. 

I made in clearer in our doc (ea37e47)